### PR TITLE
Add user message acknowledgements

### PR DESCRIPTION
**Pull request by aCinal**

### CODE REVIEW

File: common/src/common/messages.py [hunk 1]
```diff
@@ -298,6 +327,12 @@ async def cans_send(
     await socket.send(serial)
 
 
+def get_user_message_cookie(receiver: str, text: str) -> str:
```
Comment (mephi13) [inline common/src/common/messages.py:330]: why do we call it cookie? Makes no sense to me

File: client/src/client/sessions_state_machine.py [hunk 2]
```diff
@@ -189,17 +196,17 @@ def mark_outbound_session_active(self, message: CansMessage) -> None:
         # Run decryption to have the session well established at
         # olm level - without this, olm will continue producing
         # prekey messages
-        olm_message = OlmMessage(message.payload)
+        olm_message = OlmMessage(message.payload["magic"])
```
Comment (mephi13) [inline client/src/client/sessions_state_machine.py:199]: Is this "magic" only used to authenticate session at Olm level? Can't we just use "text" field? Does it have any other purpose?

Comment (mephi13) [reply client/src/client/sessions_state_machine.py:199]: ^Payload could then be a namedTuple(text, "cookie", target), which I think would be easier to use

File: common/src/common/messages.py (hunk 1 above)
Comment (mephi13) [reply common/src/common/messages.py:330]: Why not token, id, fingerprint etc? Isn't cookie a piece of data that we store clientside and later send to the server in order to have some kind of session-like behaviour in a stateless protocol such as HTTP? Maybe we do that here, Im not sure
Also, is this really unique? Is it just receiver and text? Shouldn't we add timestamp or something? It's not uncommon to have messages like "ok", "no", "yeah" etc. Is it taken into consideration in the `text` param?

File: client/src/client/sessions_state_machine.py (hunk 2 above)
Comment (aCinal) [reply client/src/client/sessions_state_machine.py:199]: Depending on message ID payload can have different structure. It cannot be typed based on user messages alone, we have other traffic also, where there are more fields in the payload. I could revert back to using message.payload of type string for the handshake messages (remove the indirection via the "magic" field)

**Review by mephi13 — APPROVED** [review]

**Review by mephi13 — APPROVED** [review]

**Pull request merged**